### DISCUSSION TITLE
Long break handling I

### DISCUSSION THREAD

**Discussion by gitthepie** [adrcotfas/goodtime#296] [Ideas]
I have my short breaks set to 5 minutes and 15-minute long breaks set to activate after every 4 sessions. But I've noticed that the long breaks aren't activating at the proper time. Either I see the long break at an unexpected time (say, after 7 sessions), or I only get 5 minute breaks; I'm not certain which one is happening because it's difficult to test without actually doing the Pomodoros. Skipping ahead by swiping right on the timer does show the breaks at the expected intervals though.

The only possibility that came to mind was the need to reset the timer on each new day, which I do anyways, but is probably already the default after the app is closed (if it isn't, perhaps we could get this as a setting to avoid the need to remember to reset?). I haven't seen any bug reports on a similar problem, so since I've already gone through the settings, I suspect the problem with getting long breaks is either due to a hidden feature (e.g. maybe I'm waiting too long after some Pomodoros) or a bug.

**Comment by adrcotfas** [comment]
How long are you waiting?

Completing a work session will increment the (invisible) work session counter only if completed in a reasonable time (work session time + break session time + 20 minutes).
f.e. For a 25 work/ 5 break and 15 minutes long break, when completing a work session the counter will increment only if the work session is finished 50 minutes or less after the last work session / else it will be set to 1.

**Reply by torotil** [reply]
I found this hidden behaviour confusing as well. For me it seemed that the long breaks were just occurring randomly.

Would it be an option to expose this session counter somewhere on the screen? Something like “X successive sessions done, next long break after Y more sessions”  would be nice.

**Reply by torotil** [reply]
For me it also seems a bit buggy on top of that: Using the above scheme 25min/5min with every 4 break being 15min, I got a 15min break after my second session.

**Comment by gitthepie** [comment]
Yes, that would explain the problem. Sometimes I do pass the invisible and unconfigurable threshold allowed.

Many potential disruptions can pass this invisible threshold. Can you add the invisible thresholds to the user settings? The hidden thresholds can be confusing for users like me who sometimes just want the app to be a simple counter of the number of work sessions and breaks.

There are situations in which I don't want to lose track of the counter but also don't want to change the break cycle. Having gone out to lunch, doing unrelated things that fall into work but don't give ready access to the phone (e.g. a period of lifting outside of desk work), being called into a meeting in which you have to work but not have ready access to your phone, using the bathroom and then getting pulled into a coworker's office before you can get back to your phone to restart your own work, delaying a break in order to finish just a bit more work but without wanting to lose the break completely, etc are examples of situations in which you still want to preserve your breaks, but also don't want to reset the Pomodoro counter.

Adding the threshold to the settings would allow the user to decide what is "reasonable" for their particular work situation or use case.

P.S. Sorry for the close and reopen. I clicked the wrong button when replying!

**Comment by adrcotfas** [comment]
Long break handling was implemented from scratch along with an optional "current session out of sessions before long break" indicator which should make things clearer. It will come in the next release.
